staging/src/kubevirt.io/api/snapshot/v1alpha1/types.go

```diff
@@ -286,6 +286,12 @@ type VirtualMachineRestoreSpec struct {
 	Target corev1.TypedLocalObjectReference `json:"target"`
 
 	VirtualMachineSnapshotName string `json:"virtualMachineSnapshotName"`
+
+	// If the target VM for the restore does not exist, it will be created. VmPatches holds JSON patches that
+	// would be applied to the VM manifest before it's created.
+	// +optional
+	// +listType=atomic
+	VmPatches []string `json:"vmPatches,omitempty"`
```

Comment:
@mhenriks 

So, this field would allow Itamar to create some per VM unique values with the clone api (like a unique mac, serial, etc...) 

How do you feel about the name VmPatches here though? The `Target` is generic right now even though practically it will only be a VM today.

What other targets might we support in the future? Is the term `VmPatches` generic enough here or should we consider a different name?

Reply:
This works for me.

We can probably support VMI targets someday.  If so, we can have `vmiPatches`.

Reply:
Maybe simply `patches` can be good enough?
I can document that the patches are expected to fit the target's type, then it's the user responsibility to provide valid patches.

WDYT?

Reply:
I've updated the PR. Thanks a lot for all of the discussions and good suggestions :)

Please share if the update is fine by you / if there's another feedback before we can merge this.